packages/cactus-plugin-keychain-aws-sm/src/main/typescript/webservices/set-keychain-entry-endpoint-v1.ts

```diff
@@ -0,0 +1,76 @@
+import { Express } from "express";
+
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+  IAsyncProvider,
+} from "@hyperledger/cactus-common";
+import {
+  IEndpointAuthzOptions,
+  IExpressRequestHandler,
+  IWebServiceEndpoint,
+} from "@hyperledger/cactus-core-api";
+import { registerWebServiceEndpoint } from "@hyperledger/cactus-core";
+
+import OAS from "../../json/openapi.json";
+
+export interface ISetKeychainEntryEndpointV1Options {
+  logLevel?: LogLevelDesc;
+}
+
+export class SetKeychainEntryEndpointV1 implements IWebServiceEndpoint {
+  public static readonly CLASS_NAME = "SetKeychainEntryEndpointV1";
+
+  private readonly log: Logger;
+
+  public get className(): string {
+    return SetKeychainEntryEndpointV1.CLASS_NAME;
+  }
+
+  constructor(public readonly options: ISetKeychainEntryEndpointV1Options) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+    this.log.debug(`Instantiated ${this.className} OK`);
+  }
+
+  private getOperation() {
+    return OAS.paths[
+      "/api/v1/plugins/@hyperledger/cactus-plugin-keychain-aws-sm/set-keychain-entry"
+    ].post;
+  }
+
+  getAuthorizationOptionsProvider(): IAsyncProvider<IEndpointAuthzOptions> {
+    // TODO: make this an injectable dependency in the constructor
+    return {
+      get: async () => ({
+        isProtected: true,
+        requiredRoles: [],
+      }),
+    };
+  }
+
+  public async registerExpress(
+    expressApp: Express,
+  ): Promise<IWebServiceEndpoint> {
+    await registerWebServiceEndpoint(expressApp, this);
+    return this;
+  }
+
+  public getVerbLowerCase(): string {
+    return this.getOperation()["x-hyperledger-cactus"].http.verbLowerCase;
+  }
+
+  public getPath(): string {
+    return this.getOperation()["x-hyperledger-cactus"].http.path;
+  }
+
+  public getExpressRequestHandler(): IExpressRequestHandler {
+    throw new Error("Method not implemented.");
```

Comment:
@jagpreetsinghsasan Please create a follow-up issue to finish the endpoint's request handler as well + write tests that use the API client instead of the plugin instance directly.

Reply:
Created #967